### ISSUE TITLE
service module is RHLE7 is not disabling service

### ISSUE THREAD

**Issue by pepee16** [ansible/ansible-modules-core#1136]
While this was working in ansible version 1.8 it is not working since  1.9.

For example following code will not disable the services which are enabled and for every item it is finished like OK (like it is in desired state already):

```
- name: DISABLE CLOUD-INIT SERVICES WE DON'T NEED.
  service: name={{ item }}
               enabled=no
  with_items: ["cloud-init-local","cloud-init","cloud-config","cloud-final"]
  when: action == "off"
  sudo: yes
```

---

From module source service.py...

```
        #
        # Systemd's systemctl
        #
        if self.enable_cmd.endswith("systemctl"):
            if self.enable:
                action = 'enable'
            else:
                action = 'disable'

            # Check if we're already in the correct state
            service_enabled = self.get_systemd_service_enabled()
            if self.enable and service_enabled:
                self.changed = False
            elif not self.enable and not service_enabled:
                self.changed = False
            elif not self.enable:
                self.changed = False

            if not self.changed:
                return
```

---

when I commented out the lines

```
           # elif not self.enable:
           #     self.changed = False
```

it is working as needed, but I'm not sure if it will not break something other. But on the other hand, changed = False should be just in case actual and desired are the same, otherwise the status should be changed so the commented lines should not be there, correct me if I'm wrong.

Anyway, thanks for any quick fix.

**Comment by bcoca** [comment]
that whole if is ... ifffy

**Comment by bcoca** [comment]
^ test with that PR

**Comment by pepee16** [comment]
I've tested your fix in my playbook and it works fine. 
When it will be available in Ansible version 1.9.x?

**Comment by bcoca** [comment]
1.9.1
